### ISSUE TITLE
How to Map Query Results to DTO in Joiner Framework with Kotlin

### ISSUE THREAD

**Issue by ByteByteBrew** [encircled/Joiner#18]
Could you please provide guidance on how to map query results to a DTO object in the Joiner framework? I couldn't find examples in the documentation that cover this scenario. I've attempted to use Querydsl's Projections for this purpose, but I encountered issues. Is there a limitation or specific method within the Joiner framework for handling result projection, especially when Projections from Querydsl are involved? Any advice or examples for this use case would be much appreciated.

**Comment by encircled** [comment]
Sorry for the late response.

As of now, there is no support for mapping projections to a DTO, only to JPA entities. As a workaround, you may map result to a Tuple and then use custom mapping to any object- https://github.com/encircled/Joiner/tree/master?tab=readme-ov-file#result-projection

I will consider implementing a native support, as it is supported by querydsl

**Comment by encircled** [comment]
Actually is does work, here is an example:

```java
    @Test
    public void toDto() {
        List<MyDto> dto = joiner.find(
                Q.select(Projections.constructor(MyDto.class,
                                QUser.user.id,
                                QUser.user.name))
                        .from(QUser.user)
        );
    }

    public static class MyDto {
        public Long id;
        public String name;

        public MyDto(Long id, String name) {
            this.id = id;
            this.name = name;
        }
    }
```

I will just add some syntax sugar for such mappings in future release

**Comment by encircled** [comment]
@ByteByteBrew released ver 1.5 with syntax like:
```kotlin
val dto = joinerKt.find(
            listOf(user.id, user.name)
                    mappingTo IdAndName::class
                    from user
        )

data class IdAndName(val id: Long, val name: String)
```
